go/vt/vtgate/endtoend/vstream_test.go

```diff
@@ -352,6 +352,12 @@ func TestVStreamCopyResume(t *testing.T) {
 					}
 					printEvents(evs) // for debugging ci failures
 				}
+				if ev.Type == binlogdatapb.VEventType_VGTID {
+					// Validate that the vgtid event the client receives from the vstream copy has a complete TableLastPK proto message.
+					// Also, to ensure that the client can resume properly, make sure that
+					// the Fields value is present in the sqltypes.Result field and not missing.
+					require.Regexp(t, `type:VGTID vgtid:{shard_gtids:{keyspace:"ks" shard:"-80" gtid:".+" table_p_ks:{table_name:"t1_copy_resume" lastpk:{fields:{name:"id1" type:INT64} rows:{lengths:1 values:"[0-9]"}}}} shard_gtids:{keyspace:"ks" shard:"80-" gtid:".+" table_p_ks:{table_name:"t1_copy_resume" lastpk:{fields:{name:"id1" type:INT64} rows:{lengths:1 values:"[0-9]"}}}}} keyspace:"ks" shard:"(80-|-80)"`, ev.String())
```

Comment:
At commit [8a0660f](https://github.com/vitessio/vitess/pull/12623/commits/8a0660fccf0945d08b69ae655c3b4da5e66aaa0f), this line failed locally, producing the following error message:

```
vstream_test.go:359:
                Error Trace:    /Users/yoheimuta/.go/vitess/go/vt/vtgate/endtoend/vstream_test.go:359
                Error:          Expect "type:VGTID vgtid:{shard_gtids:{keyspace:"ks" shard:"-80" gtid:"MySQL56/41a11ed6-c21c-11ed-b54c-c51002d8388c:1-53" table_p_ks:{table_name:"t1_copy_resume" lastpk:{fields:{name:"id1" type:INT64} rows:{lengths:1 values:"4"}}}} shard_gtids:{keyspace:"ks" shard:"80-" gtid:"MySQL56/41a11ed6-c21c-11ed-b54c-c51002d8388c:1-53" table_p_ks:{table_name:"t1_copy_resume" lastpk:{rows:{lengths:1 values:"8"}}}}} keyspace:"ks" shard:"80-"" to match "type:VGTID vgtid:{shard_gtids:{keyspace:"ks" shard:"-80" gtid:".+" table_p_ks:{table_name:"t1_copy_resume" lastpk:{fields:{name:"id1" type:INT64} rows:{lengths:1 values:"[0-9]"}}}} shard_gtids:{keyspace:"ks" shard:"80-" gtid:".+" table_p_ks:{table_name:"t1_copy_resume" lastpk:{fields:{name:"id1" type:INT64} rows:{lengths:1 values:"[0-9]"}}}}} keyspace:"ks" shard:"(80-|-80)""
                Test:           TestVStreamCopyResume
--- FAIL: TestVStreamCopyResume (1.03s)
FAIL
```

Reply:
I have added a test code here to verify that the vgtid event contains the Fields value. Although I thought it might be better to ensure that the client can use this value as a request parameter to resume, I focused on brevity rather than thoroughness. Please let me know what you think.